### PR TITLE
[Merged by Bors] - feat: port GroupTheory/GroupAction/Embedding

### PR DESCRIPTION
**Pull request by winston-h-zhang**
Everything seems to compile. This is my first port so I may be unaware of some style/comment conventions. Please, do let me know about these or any tips/tricks there are for porting! Thanks.

### CODE REVIEW

File: Mathlib/GroupTheory/GroupAction/Embedding.lean [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  rfl
+#align function.embedding.smul_def Function.Embedding.smul_def
+
+@[simp, to_additive]
+theorem smul_apply [Group G] [MulAction G β] (g : G) (f : α ↪ β) (a : α) : (g • f) a = g • f a :=
+  rfl
+#align function.embedding.smul_apply Function.Embedding.smul_apply
+
+@[to_additive]
+theorem coe_smul [Group G] [MulAction G β] (g : G) (f : α ↪ β) : ⇑(g • f) = g • f :=
```
Comment (Ruben-VandeVelde) [inline Mathlib/GroupTheory/GroupAction/Embedding.lean:46]: Perhaps this fixes the lint:
```suggestion
theorem coe_smul [Group G] [MulAction G β] (g : G) (f : α ↪ β) : ⇑(g • f) = g • ⇑f :=
```

File: Mathlib/GroupTheory/GroupAction/Embedding.lean [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import Mathlib.GroupTheory.GroupAction.Group
+import Mathlib.GroupTheory.GroupAction.Pi
+
+/-!
+# Group actions on embeddings
+
+This file provides a `mul_action G (α ↪ β)` instance that agrees with the `mul_action G (α → β)`
+instances defined by `pi.mul_action`.
+
+Note that unlike the `pi` instance, this requires `G` to be a group.
```
Comment (Ruben-VandeVelde) [inline Mathlib/GroupTheory/GroupAction/Embedding.lean:20]:
```suggestion
This file provides a `MulAction G (α ↪ β)` instance that agrees with the `MulAction G (α → β)`
instances defined by `Pi.mulAction`.

Note that unlike the `Pi` instance, this requires `G` to be a group.
```

File: Mathlib/GroupTheory/GroupAction/Embedding.lean [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+-/
+
+
+universe u v w
+
+variable {G G' α β : Type _}
+
+namespace Function.Embedding
+
+@[to_additive Function.Embedding.hasVadd]
```
Comment (Ruben-VandeVelde) [inline Mathlib/GroupTheory/GroupAction/Embedding.lean:30]:
```suggestion
@[to_additive Function.Embedding.vadd]
```

File: Mathlib/GroupTheory/GroupAction/Embedding.lean [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+instance [Group G] [MulAction G β] : SMul G (α ↪ β) :=
```
Comment (Ruben-VandeVelde) [inline Mathlib/GroupTheory/GroupAction/Embedding.lean:31]:
```suggestion
instance smul [Group G] [MulAction G β] : SMul G (α ↪ β) :=
```

**Review by Ruben-VandeVelde — APPROVED** [review]

**Comment by winston-h-zhang** [comment]
I don't understand what is happening with `lake-manifest`. With the current `lakefile` running `lake update` desyncs the `lake-manifest` with the one on `master`. Not sure which one is supposed to be correct

File: Mathlib/GroupTheory/GroupAction/Embedding.lean [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+@[to_additive]
+instance smul [Group G] [MulAction G β] : SMul G (α ↪ β) :=
+  ⟨fun g f => f.trans (MulAction.toPerm g).toEmbedding⟩
+
+@[to_additive]
+theorem smul_def [Group G] [MulAction G β] (g : G) (f : α ↪ β) :
+    g • f = f.trans (MulAction.toPerm g).toEmbedding :=
+  rfl
+#align function.embedding.smul_def Function.Embedding.smul_def
```
Comment (dupuisf) [inline Mathlib/GroupTheory/GroupAction/Embedding.lean:38]:
```suggestion
@[to_additive]
theorem smul_def [Group G] [MulAction G β] (g : G) (f : α ↪ β) :
    g • f = f.trans (MulAction.toPerm g).toEmbedding :=
  rfl
#align function.embedding.smul_def Function.Embedding.smul_def
#align function.embedding.vadd_def Function.Embedding.vadd_def
```

Comment (dupuisf) [reply Mathlib/GroupTheory/GroupAction/Embedding.lean:38]: `to_additive` doesn't add `#align` statements automatically, so these must be added manually throughout the file like this. If you're not sure what `to_additive` did for a particular declaration, you can replace `to_additive` by `to_additive?` and see what it says.

**Comment by dupuisf** [comment]
I have now merged master and manually replaced the manifest by the one currently in master, so that issue should be fixed.

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**